src/common/libsubprocess/server.c

```diff
@@ -74,7 +75,12 @@ static void proc_destructor (void **item)
 static int proc_save (subprocess_server_t *s, flux_subprocess_t *p)
 {
     void *handle;
+    const char *label = flux_cmd_get_label (p->cmd);
 
+    if (label && zhashx_insert (s->labels, label, p) < 0) {
+        errno = EEXIST;
+        return -1;
+    }
     if (!(handle = zlistx_add_end (s->subprocesses, p))) {
         errno = ENOMEM;
         return -1;
```

Comment:
Need to remove label from hash if zlistx_add_end fails?

Side note: I wonder if the hash is needed?  A given subprocess server likely won't have a gazillion subprocesses, and it won't need to find one by label except when delivering a signal or later, recovering, both exceptional conditions.  :shrug:  Just asking, no strong opinion.

Reply:
The hash was the simplest way to implement the requirement that no duplicate labels for a given server are allowed. It just seemed like the right tool for the job. Are we particularly worried about memory usage here? If so, we could easily switch to a linear search.

Reply:
Nope, I was just wondering if the complexity was worth it, but I wasn't thinking about avoiding dups.  All good.